File: ios/Quietly/Models/Note.swift
import Foundation

struct Note: Codable, Identifiable {
    let id: UUID
    let userId: UUID
    let bookId: UUID
    let content: String
    let noteType: NoteType
    let pageNumber: Int?
    let createdAt: Date
    var updatedAt: Date

    // Joined book data (for Notes view)
    var book: Book?

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case bookId = "book_id"
        case content
        case noteType = "note_type"
        case pageNumber = "page_number"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case book = "books"
    }

    var formattedDate: String {
        let formatter = DateFormatter()
        formatter.dateStyle = .medium
        formatter.timeStyle = .short
        return formatter.string(from: createdAt)
    }

    var pageLabel: String? {
        guard let page = pageNumber else { return nil }
        return "p. \(page)"
    }
}

// MARK: - Insert Model
struct NoteInsert: Codable {
    let userId: UUID
    let bookId: UUID
    let content: String
    let noteType: NoteType
    let pageNumber: Int?

    enum CodingKeys: String, CodingKey {
        case userId = "user_id"
        case bookId = "book_id"
        case content
        case noteType = "note_type"
        case pageNumber = "page_number"
    }
}

// MARK: - Book Notes Group (for Notes view)
struct BookNotesGroup: Identifiable {
    let id: UUID
    let book: Book
    let notes: [Note]

    var noteCount: Int { notes.count }
}

File: android/app/src/main/java/com/quietly/app/data/model/Note.kt
package com.quietly.app.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Note(
    val id: String,
    @SerialName("user_id")
    val userId: String,
    @SerialName("user_book_id")
    val userBookId: String,
    val content: String,
    @SerialName("note_type")
    val noteType: NoteType = NoteType.NOTE,
    @SerialName("page_number")
    val pageNumber: Int? = null,
    @SerialName("created_at")
    val createdAt: String? = null,
    @SerialName("updated_at")
    val updatedAt: String? = null,
    // Joined data
    @SerialName("user_book")
    val userBook: UserBook? = null
)
